### PR TITLE
Create `<define-assembly name="impact">` (fix #1129)

### LINKED ISSUE #1129: Create an `#assembly_oscal-ssp_impact` SSP model
# User Story:

Currently, inside the `#assembly_oscal-ssp_system-information` model (within the SSP schema) there are 3 fields which implement the same object. The fields are: 
- `confidentiality-impact`
- `integrity-impact`
- `availability-impact`

In order to avoid duplication, I suggest creating a new `#assembly_oscal-ssp_impact` model within the SSP schema to define the `impact` component
The model will be used by the 3 fields mentioned, and will deprecate the repeated implementation within the fields.

Since the fields themselves remain the same (and just the way they are defined in the schema is changed), this proposed change is backwards compatible and can be implemented in OSCAL `v1.x.y`.

## Goals:

Reduce duplication in the SSP schema by creating a new `#assembly_oscal-ssp_impact` model.

## Acceptance Criteria

- [ ] All [OSCAL website](https://pages.nist.gov/OSCAL) and readme documentation affected by the changes in this issue have been updated. Changes to the OSCAL website can be made in the docs/content directory of your branch.
- [ ] A Pull Request (PR) is submitted that fully addresses the goals of this User Story. This issue is referenced in the PR.
- [ ] The CI-CD build process runs without any reported errors on the PR. This can be confirmed by reviewing that all checks have passed in the PR.

### PR DESCRIPTION
**Pull request by guyzyl**
# Committer Notes

This PR fixes issue #1129 .
See issue for the reasoning (and discussion) behind this PR.

### All Submissions:

- [x] Have you selected the correct base branch per  [Contributing](https://github.com/usnistgov/OSCAL/blob/master/CONTRIBUTING.md) guidance?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/usnistgov/OSCAL/pulls) for the same update/change?
- [x] Have you squashed any non-relevant commits and commit messages? \[[instructions](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History)\]
- [x] Do all automated CI/CD checks pass?

### CODE REVIEW

File: src/metaschema/oscal_ssp_metaschema.xml [hunk 1]
```diff
@@ -317,6 +281,21 @@
       </allowed-values>
     </constraint>
   </define-assembly>
+  <define-assembly name="impact">
+    <formal-name>Impact Level</formal-name>
+    <description>The expected level of impact resulting from the described information.</description>
```
Comment (aj-stein-nist) [inline src/metaschema/oscal_ssp_metaschema.xml:286]: So one of the things we will lose from the more modular code is the same `description` for the three different subsets of impact level. Personally, this does not bother me, but I think in terms of importance for that unique documentation per impact level is something we lose with the `assembly/@ref` approach. Are we cool with that at a high-level, @david-waltermire-nist, if I am happy with the other changes made here?

They look good so far. Will test locally on my workstation and have a deep think about other implications, but it seems low risk, good reward.

**Review by aj-stein-nist — COMMENTED** [review]
I might want to circle back with the team on the docs, but looks good so far. I will test it locally more more rigorously than just eyeballing it, and come back with formal approval or to request changes. Still want this question cleared up in advance.

**Review by wendellpiez — APPROVED** [review]
The `use-name` feature may not have been available when this was originally sketched - nice find, @guyzyl

**Comment by guyzyl** [comment]
@wendellpiez thanks! But the credit goes to @david-waltermire-nist for [pointing it out](https://github.com/usnistgov/OSCAL/issues/1129#issuecomment-1066135260) to me.

**Comment by david-waltermire-nist** [comment]
The one problem with this change is that the formal names for the individual impact metrics will be normalized. I'd like to add a feature to Metaschema that will allow the model references to override the formal names, preserving the current naming for the documentation.

I am going to push this to 1.1.0 to allow time to make this refinement.

**Comment by david-waltermire-nist** [comment]
The necessary change to the Metaschema model are in usnistgov/metaschema#197.

See https://github.com/usnistgov/metaschema/blob/0b61192f3559218b7097b86deeeac837794f5388/schema/xml/metaschema.xsd#L445

The changes to complete this PR can be made once this PR is merged and issue #1338 is resolved.

File: src/metaschema/oscal_ssp_metaschema.xml (hunk 1 above)
Comment (david-waltermire-nist) [reply src/metaschema/oscal_ssp_metaschema.xml:293]: I added the formal-names and descriptions back since this was added as a Metaschema feature.

**Review by david-waltermire-nist — APPROVED** [review]
@guyzyl Thanks! Finally completed this.

**Comment by guyzyl** [comment]
@david-waltermire-nist really glad to finally see this merged 🙂

**Pull request merged**